Source/DafnyCore/CompileNestedMatch/MatchFlattener.cs

```diff
@@ -132,21 +96,21 @@ public MatchFlattener(ErrorReporter reporter, FreshIdGenerator idGenerator)
 
     CaseBody compiledMatch = CompilePatternPaths(state, new HoleCtx(), LinkedLists.Create(nestedMatchExpr.Source), paths);
     if (compiledMatch is null) {
-      // Happens only if the match has no cases, create a Match with no cases as resolved expression and let ResolveMatchExpr handle it.
-      var result = new MatchExpr(nestedMatchExpr.tok, nestedMatchExpr.Source, new List<MatchCaseExpr>(), nestedMatchExpr.UsesOptionalBraces);
-      result.Type = nestedMatchExpr.Type;
-      FillMissingCases(result);
-      return result;
-    } else if (compiledMatch.Node is Expression expression) {
+      Reporter.Error(MessageSource.Resolver, nestedMatchExpr.Tok, "match has no cases");
```

Comment:
Note that this code is about expressions not statements. I used a resolution error here because it doesn't seem right to pass an empty match expression to the downstream code, and the previous behavior also gave a resolution error in this scenario, but I'll use `assert false; var x :| true; x`